cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsUpdateService.java

```diff
@@ -121,4 +136,43 @@ private CloudResource getCloudFormationStack(List<CloudResource> resources) {
         return resources.stream().filter(resource -> ResourceType.CLOUDFORMATION_STACK == resource.getType()).findFirst()
                 .orElseThrow(() -> new NotFoundException("CloudFormation stack is not found, the resource might have been deleted."));
     }
+
+    public void modifyVolumes(AuthenticatedContext authenticatedContext, List<String> volumeIds, String diskType, int size) throws Exception {
+        AmazonEc2Client amazonEC2Client = getEc2Client(authenticatedContext);
+        for (String volume: volumeIds) {
+            LOGGER.debug("Modifying volume {} to disk type {} and size {}", volume, diskType, size);
+            try {
+                ModifyVolumeRequest modifyVolumeRequest = ModifyVolumeRequest.builder().volumeId(volume).volumeType(diskType).size(size).build();
+                amazonEC2Client.modifyVolume(modifyVolumeRequest);
+            } catch (software.amazon.awssdk.services.ec2.model.Ec2Exception ex) {
+                LOGGER.error(String.format("AWS threw BAD Request exception: %s", ex.getMessage()));
+                if (ex.getMessage().startsWith("You've reached the maximum modification rate per volume limit.")) {
+                    continue;
+                }
```

Comment:
@sodre90 The only reason this error with that specific message is thrown is because a modification operation was done previously within six hours - that is the limit AWS has set. i handled it this way because if the previous flow has successfully modified the volume but failed somewhere else in the flow, i cannot make the client wait 6 hours before they can retry

Reply:
What happens if I want to update the disk 2 times in an hour? first time I modify it to 1Tb, then to 2Tb, will it be successful and the disks will be modified to 2 Tb?

Reply:
you cannot do it with AWS. the disks cannot be modified within 6 hours. if the client first tries for 2 TB and its successful, and tries to do it for 6 TB within 2 hours, the first resizing on CP side will fail, and the host orchestrator will fail as well - and the entire flow will fail. The client will be notified why it failed on the Events window, so he can wait until the open time to retry